src/Chainweb/Mempool/InMem.hs

```diff
@@ -454,21 +455,22 @@ recordRecentTransactions maxNumRecent newTxs rlog = rlog'
     numNewItems = V.length newTxs
     oldNext = _rlNext rlog
     newNext = oldNext + fromIntegral numNewItems
-    newTxs' = reverse ([oldNext..] `zip` V.toList newTxs)
-    newL' = newTxs' ++ _rlRecent rlog
-    newL = force $ take maxNumRecent newL'
+    newTxs' = V.reverse (V.map (T2 oldNext) newTxs)
+    newL' = newTxs' <> _rlRecent rlog
+    newL = force $ V.take maxNumRecent newL'
 
 
 -- | Get the recent transactions from the transaction log. Returns Nothing if
 -- the old high water mark is too out of date.
 getRecentTxs :: Int -> MempoolTxId -> RecentLog -> Maybe [TransactionHash]
 getRecentTxs maxNumRecent oldHw rlog
     | oldHw <= oldestHw || oldHw > oldNext = Nothing
-    | oldHw == oldNext = Just []
-    | otherwise = Just $! txs
+    | oldHw == oldNext = Just mempty
+    | otherwise = Just $! V.toList txs
 
   where
     oldNext = _rlNext rlog
     oldestHw = oldNext - fromIntegral maxNumRecent
-    txs = map snd $ takeWhile pred $ _rlRecent rlog
-    pred x = fst x >= oldHw
+    txs = V.map (\(T2 _ b) -> b) $ V.takeWhile pred $ _rlRecent rlog
```

Comment:
I've added strict `fst` and strict `snd` combinators here: https://github.com/kadena-io/chainweb-node/blob/master/src/Chainweb/Utils.hs#L1099

```suggestion
    txs = V.map ssnd $ V.takeWhile pred $ _rlRecent rlog
```

Reply:
nice.